### DISCUSSION TITLE
Allow to adjust the ES_JAVA_OPTS via compose.yaml

### DISCUSSION THREAD

**Discussion by 4y55** [nextcloud/all-in-one#5311] [Ideas] [answered]
The Elasticsearch instance is crashing due to an OutOfMemoryError caused by insufficient Java heap space. The logs indicate that the heap size is currently set to 512MB:
[logs.txt](https://github.com/user-attachments/files/17089869/logs.txt)

How do I increase ES_JAVA_OPTS?

**Comment by szaimen** [comment]
Hi, you currently cannot but we will incrase the default with https://github.com/nextcloud/all-in-one/pull/5323.

However a real fix is to make this configurable. I'll add this to https://github.com/nextcloud/all-in-one/issues/5251

See https://github.com/nextcloud/all-in-one/pull/793 as reference

**Reply by 4y55** [reply]
Thanks for your attention.

I would like a current solution, but I will wait for the implementation.

Increasing the default values ​​may require more robust requirements for the host machine.

I believe the best way forward is to allow the values ​​to be changed manually via compose.

Anyway, thanks.

**Comment by szaimen** [comment, marked as answer]
This will be done in https://github.com/nextcloud/all-in-one/pull/6005

**Reply by szaimen** [reply]
This is now released with v10.6.0 Beta. Testing and feedback is welcome! See https://github.com/nextcloud/all-in-one#how-to-switch-the-channel
